sklearn/utils/estimator_checks.py

```diff
@@ -631,6 +632,45 @@ def check_sample_weights_invariance(name, estimator_orig):
                                         % name)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_sample_weights_equivalence_sampling(name, estimator_orig):
+    # check that the estimators yield same results for
+    # over-sample dataset by indice filtering and using sample_weight
+    if (has_fit_parameter(estimator_orig, "sample_weight") and
+            not (hasattr(estimator_orig, "_pairwise")
+                 and estimator_orig._pairwise)):
+        # We skip pairwise because the data is not pairwise
+
+        estimator1 = clone(estimator_orig)
+        estimator2 = clone(estimator_orig)
+        set_random_state(estimator1, random_state=0)
+        set_random_state(estimator2, random_state=0)
+
+        if is_classifier(estimator1):
+            X, y = load_iris(return_X_y=True)
+        else:
+            X, y = load_boston(return_X_y=True)
+        y = enforce_estimator_tags_y(estimator1, y)
+
+        indices = np.arange(start=0, stop=y.size, step=2)
+        sample_weight = np.ones((y.size,)) * np.bincount(indices,
```

Comment:
I find this hard to read. I think you are just doing
```
weight = zeros (...)
weight[::2] = 1
```

But maybe you wanted this to be flexible to setting indices with random.choice. 

I don't think that multiplying ones by something is a very useful operation though.

Reply:
I copy paste some code but simplify the indices. Your proposal is better.